Client/Frontend/Browser/TopTabsViewController.swift

```diff
@@ -0,0 +1,714 @@
+    }
+    
+    required init?(coder aDecoder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+}
+
+class TopTabsBackgroundDecorationView : UICollectionReusableView {
+    static let Identifier = "TopTabsBackgroundDecorationViewIdentifier"
+    private lazy var right = SingleCurveView(right: true)
+    private lazy var left = SingleCurveView(right: false)
+    
+    lazy var centerBackground: UIView = {
+        let centerBackground = UIView()
+        centerBackground.backgroundColor = TopTabsUX.TopTabsBackgroundNormalColorInactive
+        return centerBackground
+    }()
+    
+    override init(frame: CGRect) {
+        super.init(frame: frame)
+        
+        self.addSubview(right)
+        self.addSubview(left)
+        self.addSubview(centerBackground)
+        
+        right.snp_makeConstraints { make in
+            make.right.equalTo(self)
+            make.top.equalTo(self)
+            make.bottom.equalTo(self)
+            make.width.equalTo(SingleCurveView.CurveWidth)
+        }
+        left.snp_makeConstraints { make in
+            make.left.equalTo(self)
+            make.top.equalTo(self)
+            make.bottom.equalTo(self)
+            make.width.equalTo(SingleCurveView.CurveWidth)
+        }
+        centerBackground.snp_makeConstraints { make in
+            make.left.equalTo(left.snp_right)
+            make.right.equalTo(right.snp_left)
+            make.top.equalTo(self)
+            make.bottom.equalTo(self)
+        }
+    }
+    
+    required init?(coder aDecoder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+    
+    override func layoutSubviews() {
+        super.layoutSubviews()
+        setNeedsDisplay()
+    }
+    
+    override func applyLayoutAttributes(layoutAttributes: UICollectionViewLayoutAttributes) {
+        super.applyLayoutAttributes(layoutAttributes)
+        self.setNeedsDisplay()
+    }
+    
+    override func prepareForReuse() {
+        super.prepareForReuse()
+        setNeedsDisplay()
+    }
+    
+    override func drawRect(rect: CGRect) {
+        super.drawRect(rect)
+    }
+    
+    private class SingleCurveView: UIView {
+        static let CurveWidth: CGFloat = 50
+        var right: Bool = true
+        init(right: Bool) {
+            self.right = right
+            super.init(frame: CGRectZero)
+        }
+        
+        required init?(coder aDecoder: NSCoder) {
+            fatalError("init(coder:) has not been implemented")
+        }
+        
+        override func drawRect(rect: CGRect) {
+            super.drawRect(rect)
+            
+            let fillColor = TopTabsUX.TopTabsBackgroundNormalColorInactive
+            
+            let width = frame.width
+            let height = frame.height
+            let x1: CGFloat = 32.84
+            let x2: CGFloat = 5.1
+            let x3: CGFloat = 19.76
+            let x4: CGFloat = 58.27
+            let x5: CGFloat = -12.15
```

Comment:
This looks like the same code and points for the `BezierView`. Is there a way we can DRY this up instead of having it in multiple places?